Bottom: 7dc2e83797b5b3d45c42b1ec23c6eb4ab8451e3e
Top:    c61b24147b53d5633133ac638ef68a5ff802b44d
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-19 15:19:56 +0800

Refresh of autofs-5.0.7-allow-use-of-hosts-map-in-maps.patch

---

diff --git a/modules/parse_sun.c b/modules/parse_sun.c
index f6e79b7..f7cca85 100644
--- a/modules/parse_sun.c
+++ b/modules/parse_sun.c
@@ -1442,17 +1442,18 @@ int parse_mount(struct autofs_point *ap, const char *name,
 			error(LOGOPT_ANY, "options %s", options);
 
 			l = parse_mapent(p, options, &myoptions, &loc, ap->logopt);
-			if (!l &&
-			    !(strstr(options, "fstype=autofs") &&
-			      strstr(options, "hosts"))) {
-				error(LOGOPT_ANY, "I think I'm a hosts map? l %d", l);
-				cache_delete_offset_list(mc, name);
-				cache_multi_unlock(me);
-				cache_unlock(mc);
-				free(path);
-				free(options);
-				pthread_setcancelstate(cur_state, NULL);
-				return 1;
+			if (!l) {
+				if (!(strstr(myoptions, "fstype=autofs") &&
+				      strstr(myoptions, "hosts"))) {
+					error(LOGOPT_ANY, "I think I'm a hosts map? l %d", l);
+					cache_delete_offset_list(mc, name);
+					cache_multi_unlock(me);
+					cache_unlock(mc);
+					free(path);
+					free(options);
+					pthread_setcancelstate(cur_state, NULL);
+					return 1;
+				}
 			}
 
 			p += l;
